src/main/java/se/isselab/HAnS/metricsView/MetricsViewFactory.java

```diff
@@ -0,0 +1,160 @@
+        toolWindow.setIcon(AnnotationIcons.PluginIcon);
+        createButton(project, toolWindow);
+        addContent(toolWindow);
+    }
+
+    private void createButton(Project project, ToolWindow toolWindow) {
+        triggerButton = new JButton("Refresh Metrics");
+        triggerButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                // Run the task in the background
+                ProgressManager.getInstance().run(new Task.Backgroundable(project, "Refreshing metrics") {
+                    @Override
+                    public void run(@NotNull ProgressIndicator indicator) {
+                        refreshTableContent(project, toolWindow);
+                    }
+                });
+            }
+        });
+    }
+
+    private void addContent(ToolWindow toolWindow) {
+        contentPanel = new JPanel();
+        contentPanel.setLayout(new BorderLayout());
+
+        JPanel buttonPanel = new JPanel();
+        buttonPanel.add(triggerButton);
+        contentPanel.add(buttonPanel, BorderLayout.NORTH);
+
+        ContentFactory contentFactory = ContentFactory.getInstance();
+        Content content = contentFactory.createContent(contentPanel, "", false);
+        toolWindow.getContentManager().addContent(content);
+    }
+
+    private void refreshTableContent(Project project, ToolWindow toolWindow) {
+        DefaultTableModel tableModel = new DefaultTableModel(new Object[]{"Feature", "Scattering Degree", "Tangling Degree", "Line Count"}, 0) {
+            @Override
+            public Class<?> getColumnClass(int columnIndex) {
+                // Set appropriate classes for sorting
+                switch (columnIndex) {
+                    case 1: // Scattering Degree column
+                    case 2: // Tangling Degree column
+                    case 3: // Line Count column
+                        return Integer.class; // Use Integer class for numerical sorting
+                    default:
+                        return Object.class; // Default class
+                }
+            }
+        };
+
+        JBTable table = createTable(tableModel);
+
+        FeatureService featureService = project.getService(FeatureService.class);
+        List<FeatureModelFeature> features = featureService.getFeatures();
+        List<FeatureModelFeature> root = featureService.getRootFeatures();
+
+        populateTable(project, table, features, root, featureService);
+
+        table.setDefaultRenderer(Object.class, new IntegerTableCellRenderer());
+        table.setAutoCreateRowSorter(true);
+        JBScrollPane scrollPanel = new JBScrollPane(table);
+
+        contentPanel.add(scrollPanel, BorderLayout.CENTER);
+        contentPanel.revalidate();
+        contentPanel.repaint();
+    }
+
+    private JBTable createTable(DefaultTableModel tableModel) {
```

Comment:
I would prefer if this method was only for Creating the sorters. The table creation can be in the original method.